component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -1522,4 +1535,29 @@ private void setUserClaimValues(AuthenticatedUser authenticatedUser, Map<String,
             throw new AuthenticationFailedException(errorMessage, e);
         }
     }
+
+    private String getUnlockTime(AuthenticatedUser authenticatedUser)
+            throws AuthenticationFailedException {
+
+        String username = authenticatedUser.toFullQualifiedUsername();
+        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+        String unlockTime = null;
+        try {
+            UserRealm userRealm = getUserRealm(username);
+            if (userRealm == null) {
+                throw new AuthenticationFailedException("UserRealm is null for user : " + username);
+            }
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            if (userStoreManager != null) {
```

Comment:
Shall we change this logic to check `null` and add a debug log?

Reply:
fixed in f3b362c